### ISSUE TITLE
Images from extensions

### ISSUE THREAD

**Issue by MoppieMop** [Scarbous/mr_tinypng#1]
I tried mr_tinypng and it worked great, except for the page were the ke_search-module is located.
Following (shortened) output appears:

```
Invalid source specified (HTTP 400/JSON validation error) 
32 Tinify\Exception::create("Invalid source specified", "JSON validation error", 400)
31 Tinify\Client::request("post", "https://api.tinify.com/shrink", "{"error":"JSON validation error","message":"Invalid source specified"}")
...
28 Scarbous\MrTinypng\Service\TinypngService::shrinkImage("/home/www/typo3/typo3conf/ext/ke_search/res/img/attention.gif", "/home/www/typo3/typo3temp/attention.gif1592229915gif")
...
```

So the image of ke_search is the problem. Is it the format gif? Is it because it's an image of an extension?

**Comment by saphrus** [comment]
I had exactly the same problem. For the reason of file_get_contents being blocked on our system, I had to extend some classes anyway So I wrapped everything happening in TinypngService->shrinkImage within 
`if ((pathinfo($source, PATHINFO_EXTENSION) == 'png') || (pathinfo($source, PATHINFO_EXTENSION) == 'jpg'))`

**Comment by Scarbous** [comment]
**At first, thank you for the feedback.**

@MoppieMop
The Problem is, that Tinypng can't shrink GIFs.
I will add a mimetype condition.

@saphrus
You say: "I had to extend some classes anyway"
If you send me your changes, maybe I can add it to the extension.

**Comment by Scarbous** [comment]
Fixed through #2

**Comment by Scarbous** [comment]
Fixed through #2
